### PR TITLE
fix: fix Button findDOMNode error

### LINKED ISSUE #48809: [Button] Button-findDOMNode
### Reproduction link

[Edit on StackBlitz](https://stackblitz.com/edit/react-zbmzv8-cdwnta?file=demo.tsx)

### Steps to reproduce

点击按钮 查看控制台

### What is expected?

无

### What is actually happening?

无

| Environment | Info |
| --- | --- |
| antd | 5.17.0 |
| React | 18.3.1 |
| System | Windows |
| Browser | Chrome |

---
无

### PR DESCRIPTION
**Pull request by wanpan11**
[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

fix https://github.com/ant-design/ant-design/issues/48809

### 💡 Background and solution

Button 组件点击时 WaveEffect 组件没有使用 setNodeRef 

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |      fix Button findDOMNode error     |
| 🇨🇳 Chinese |      修复 Button 组件 findDOMNode 报错    |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by li-jia-nan** [comment]
加个测试用例吧

File: components/_util/wave/WaveEffect.tsx [hunk 1]
```diff
@@ -125,9 +125,12 @@ const WaveEffect: React.FC<WaveEffectProps> = (props) => {
         return false;
       }}
     >
-      {({ className: motionClassName }) => (
+      {({ className: motionClassName }, setNodeRef) => (
```
Comment (zombieJ) [inline components/_util/wave/WaveEffect.tsx:128]: 补一个 test case~

Comment (li-jia-nan) [reply components/_util/wave/WaveEffect.tsx:128]: 是不是可以用 `composeRef` ？

Comment (li-jia-nan) [reply components/_util/wave/WaveEffect.tsx:128]:
```tsx
{({ className: motionClassName }, ref) => (
  <div
    ref={composeRef(divRef, ref)}
    className={classNames(className, { 'wave-quick': isSmallComponent }, motionClassName)}
    style={waveStyle}
  />
)}
```

Comment (wanpan11) [reply components/_util/wave/WaveEffect.tsx:128]: 好的 我还在想测试用例咋写🤣

Comment (li-jia-nan) [reply components/_util/wave/WaveEffect.tsx:128]: 差不多这样？

```ts
const errorSpy = jest.spyOn(console, 'error').mockImplementation(() => {});
……
expect(errorSpy).not.toHaveBeenCalled();
```

Comment (wanpan11) [reply components/_util/wave/WaveEffect.tsx:128]: 这个好像现有的快照就会有这个报错提示，我写了个别的 佬们康康

File: components/_util/__tests__/wave.test.tsx [hunk 2]
```diff
@@ -14,6 +15,12 @@ jest.mock('rc-util/lib/Dom/isVisible', () => {
   return mockFn;
 });
 
+jest.mock('rc-motion', () => (props: CSSMotionProps) => {
+  (global as any).motionChildren = jest.fn(props.children);
+  const CSSMotion = jest.requireActual('rc-motion').default;
+  return <CSSMotion {...props}>{(global as any).motionChildren}</CSSMotion>;
+});
```
Comment (afc163) [inline components/_util/__tests__/wave.test.tsx:22]: 这个测试用例和原始问题 https://github.com/ant-design/ant-design/issues/48809 的关系是？

File: components/_util/wave/WaveEffect.tsx [hunk 3]
```diff
@@ -3,6 +3,7 @@ import classNames from 'classnames';
 import CSSMotion from 'rc-motion';
 import raf from 'rc-util/lib/raf';
 import { render, unmount } from 'rc-util/lib/React/render';
+import { composeRef } from 'rc-util/lib/ref';
```
Comment (li-jia-nan) [inline components/_util/wave/WaveEffect.tsx:6]:
```suggestion
import { composeRef } from 'rc-util/es/ref';
```

File: components/_util/__tests__/wave.test.tsx (hunk 2 above)
Comment (afc163) [reply components/_util/__tests__/wave.test.tsx:28]: 原始问题是警告的问题。

File: components/_util/__tests__/wave.test.tsx [hunk 4]
```diff
@@ -95,6 +95,9 @@ describe('Wave component', () => {
 
     expect(document.querySelector('.ant-wave')).toBeFalsy();
 
+    const errorSpy = jest.spyOn(console, 'error').mockImplementation(() => {});
+    expect(errorSpy).not.toHaveBeenCalled();
```
Comment (afc163) [inline components/_util/__tests__/wave.test.tsx:99]: 这样一定不会触发的，改不改都不会触发。

**Comment by wanpan11** [comment]
现在可以了

image

File: components/_util/__tests__/wave.test.tsx (hunk 4 above)
Comment (zombieJ) [reply components/_util/__tests__/wave.test.tsx:99]: 试了一下会触发，看了是 test 的 `render` 已经包过 StrictMode 了

**Review by zombieJ — APPROVED** [review]

**Pull request merged**